salt/tests/unit/modules/files/test_metalk8s_volumes.yaml

```diff
@@ -0,0 +1,499 @@
+  - name: my-sparse-volume
+    pillar_volumes: *volumes_details
+    has_partition: True
+    raise_msg: "backing device `/var/lib/metalk8s/storage/sparse/f1d78810-3787-4ca4-b712-50a269e42560` contains a partition table"
+
+  ## RAW BLOCK DEVICE volume
+  # format the raw block device volume in ext4
+  - name: my-raw-block-device-volume
+    pillar_volumes: *volumes_details
+    mkfs_output: |
+
+  # error when formatting the raw block device volume in ext4
+  - name: my-raw-block-device-volume
+    pillar_volumes: *volumes_details
+    raise_msg: "error while trying to run `mkfs.ext4 -F .* /dev/sda1`: An error has occurred"
+
+  # format the raw block device volume in xfs
+  - name: my-xfs-volume
+    pillar_volumes: *volumes_details
+    mkfs_output: |
+
+  # error when formatting the raw block device volume in xfs
+  - name: my-xfs-volume
+    pillar_volumes: *volumes_details
+    raise_msg: "error while trying to run `mkfs.xfs -f .* /dev/sda2`: An error has occurred"
+
+  ## Invalid volumes
+  # volume with invalid storage class name
+  - name: my-invalid-storage-class-volume
+    pillar_volumes: *volumes_details
+    raise_msg: "StorageClass invalid-storage-class not found"
+
+  # volume with storage class with invalid filesystem
+  - name: my-invalid-fstype-volume
+    pillar_volumes: *volumes_details
+    raise_msg: "unsupported filesystem: invalid-fstype"
+
+  # specified volume is not in the pillar
+  - name: unknown-volume
+    pillar_volumes: *volumes_details
+    raise_msg: volume unknown-volume not found in pillar
+
+is_cleaned_up:
+  ## SPARSE volume
+  # sparse file does not exists
+  - name: my-sparse-volume
+    pillar_volumes: *volumes_details
+    result: True
+
+  # sparse file exists and associated with a loop device
+  - name: my-sparse-volume
+    pillar_volumes: *volumes_details
+    is_provisioned: True
+    exists: True
+    result: False
+
+  # sparse file exists but not associated with any device
+  - name: my-sparse-volume
+    pillar_volumes: *volumes_details
+    is_provisioned: False
+    exists: True
+    result: False
+
+  ## RAW BLOCK DEVICE volume
+  # raw block device always cleaned up (nothing to clean)
+  - name: my-raw-block-device-volume
+    pillar_volumes: *volumes_details
+    result: True
+
+  ## Invalid volumes
+  # specified volume is not in the pillar
+  - name: unknown-volume
+    pillar_volumes: *volumes_details
+    raises: True
+    result: volume unknown-volume not found in pillar
+
+clean_up:
+  ## SPARSE volume
+  # clean up a sparse file
+  - name: my-sparse-volume
+    pillar_volumes: *volumes_details
+
+  # sparse file does not exists
+  - name: my-sparse-volume
+    pillar_volumes: *volumes_details
+    remove_error: [2, "No such file or directory", "..."]
+
+  # error when trying to remove the sparse file
+  - name: my-sparse-volume
+    pillar_volumes: *volumes_details
+    remove_error: "An error has occurred during remove"
+    raise_msg: "An error has occurred during remove"
```

Comment:
Not entirely certain from reading these example errors: is the `clean_up` method copying the error message from `os.remove` (or below, from `fcntl.ioctl`), or are we using a generic message which matches what's above? If the former, maybe it'd be nice to put some random word in the message, to make it obvious (e.g. "A wild spaghetti monster appeared!"), but if the latter, maybe we should show different messages between the actual error and what gets raised (sorry, didn't check the code).

Reply:
Right, right now I just put `An error has occurred` everywhere when I wanted to raise something but indeed it could be clearer we some random word. 
And indeed here we use the message raised by `os.remove` (and `fcntl.ioctl`)